scripts/tsc.js

```diff
@@ -0,0 +1,105 @@
+const { DiagnosticCategory } = require("typescript");
```

Comment:
I'm looking at this and I don't see why we cannot use the `tsc` CLI instead.

There's no performance benefit to running this script over CLI itself and it adds some maintainability debt. Wouldn't these 2 suffice (with `tsc` excluded from build script)?
```
    "start": "node scripts/build.js watch & npx tsc --watch --preserveWatchOutput",
    "build": "npx tsc && node scripts/build.js",
```